cpp/ql/src/Likely Bugs/Format/NonConstantFormat.ql

```diff
@@ -79,6 +73,20 @@ class ConstFlow extends DataFlow::Configuration {
   override predicate isSink(DataFlow::Node sink) {
     exists(FormattingFunctionCall fc | sink.asExpr() = fc.getArgument(fc.getFormatParameterIndex()))
   }
+
+  override predicate isAdditionalFlowStep(DataFlow::Node source, DataFlow::Node sink) {
+  	none()
+  	or
+    // an element picked from an array of string literals is a string literal
+    exists(Variable v, int a |
+      a = sink.asExpr().(ArrayExpr).getArrayOffset().getValue().toInt() and
+      v = sink.asExpr().(ArrayExpr).getArrayBase().(VariableAccess).getTarget()
+    |
+    // we disallow parameters, since they may be bound to unsafe arguments
+    // at various call sites.
+      not v instanceof Parameter and source.asExpr() instanceof StringLiteral
+      )
```

Comment:
This predicate doesn't relate `source` and `sink`, so it computes a Cartesian product. Let's discuss how to fix it in the next meeting.